src/Explorer/index.jsx

```diff
@@ -0,0 +1,554 @@
+      x => x.toLowerCase(),
+    ) === row[property].toLowerCase();
+    let filtersApplyContains = filtersApplied[property].selectedValues.filter(
+      x => row[property].toLowerCase() === x.toLowerCase(),
+    );
+    filtersApplyContains = filtersApplyContains.length > 0;
+    const filtersApply = filtersApplyMatch || filtersApplyContains;
+    if (!filtersApply) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function flattenHistograms(listOfHistograms) {
+  // This absurd function combines Guppy histograms from an arbitrary
+  // number of commons for the purpose of rendering explorer charts.
+  const flattened = {};
+
+  for (let j = 0; j < listOfHistograms.length; j += 1) {
+    // eslint-disable-next-line no-continue
+    if (!listOfHistograms[j] || typeof listOfHistograms[j] === 'undefined') continue;
+    const keys = Object.keys(listOfHistograms[j]);
+    for (let k = 0; k < keys.length; k += 1) {
+      const fieldName = keys[k];
+      if (!Object.prototype.hasOwnProperty.call(flattened, fieldName)) {
+        flattened[fieldName] = {};
+      }
+
+      if (!listOfHistograms[j][keys[k]]) {
+        continue; // eslint-disable-line no-continue
+      }
+      const histogramsForKey = listOfHistograms[j][keys[k]].histogram;
+      if (!histogramsForKey || typeof histogramsForKey === 'undefined') {
+        continue; // eslint-disable-line no-continue
+      }
+      for (let z = 0; z < histogramsForKey.length; z += 1) {
+        const fieldValue = histogramsForKey[z].key;
+        const fieldCount = histogramsForKey[z].count;
+        if (!Object.prototype.hasOwnProperty.call(flattened[fieldName], fieldValue)) {
+          flattened[fieldName][fieldValue] = fieldCount;
+        } else {
+          flattened[fieldName][fieldValue] += fieldCount;
+        }
+      }
+    }
+  }
+
+  const result = {};
+  Object.keys(flattened).forEach((key) => {
+    result[key] = { histogram: [] };
+    Object.keys(flattened[key]).forEach((subKey) => {
+      result[key].histogram.push({ key: subKey, count: flattened[key][subKey] });
+    });
+  });
+
+  return result;
+}
+
+class Explorer extends React.Component {
+  constructor(props) {
+    super(props);
+    const tabIndex = 0;
+    this.state = {
+      tab: tabIndex > 0 ? tabIndex : 0,
+      rawData: [],
+      filteredData: [],
+      paginatedData: [],
+      counts: {
+        supported_data_resource: 0,
+        dataset: 0,
+      },
+      chartData: { },
+      dataExplorerConfig: config.dataExplorerConfig,
+      datasetsCount: 0,
+      loading: true,
+    };
+    this.filterGroupRef = React.createRef();
+    this.tableRef = React.createRef();
+  }
+
+  componentWillMount() {
+    this.initializeData();
+  }
+
+  obtainSubcommonsData = (subcommonsConfig) => {
+    const subcommonsURL = subcommonsConfig.URL;
+    const subcommonsName = subcommonsConfig.name;
+    const flatModelQueryURL = 'guppy/download';
```

Comment:
feels like we should put all these hardcoded API endpoints (https://github.com/uc-cdis/data-ecosystem-portal/blob/feat/data-explorer/src/Explorer/index.jsx#L145 , https://github.com/uc-cdis/data-ecosystem-portal/blob/feat/data-explorer/src/Explorer/index.jsx#L298 and https://github.com/uc-cdis/data-ecosystem-portal/blob/feat/data-explorer/src/DatasetBrowser/index.jsx#L73) into localconf.js and import from there to keep things in a same place